### ISSUE TITLE
Shell scripts in 'tool' won't run on Windows

### ISSUE THREAD

**Issue by DanTup** [flutter/devtools#5909]
I'm trying to switch back to Windows for msot of my Dart-Code development (because ~65% of users are windows, but few pre-release/internal users are). While trying to [run local DevTools embedded in VS Code](https://github.com/flutter/devtools/blob/master/CONTRIBUTING.md#development-vs-code-integration) I found that lots of the scripts are calling shell scripts (`.sh`) and won't run on Windows.

The ones I've looked at are all fairly trivial. We could make Windows versions (`.bat` or `.ps1`) fairly easily, but it would mean maintaining two. I wonder whether they should just be converted to simple Dart scripts (or  perhaps something like [grinder](https://github.com/google/grinder.dart)) to run on any platform (although, right now the root folder doesn't have a `pubspec.yaml` so can't have dependencies).

I don't know how worthwhile this is. It might be that I'm the only person to benefit from it, and I do have easy access to a Mac. It's possible there are potential contributors that are also on Windows (without access to a Mac), but I don't know (I think you could also get some way without needing these scripts by just running from source).

This isn't blocking, for now I'll just use a Mac when working on VS Code integration, mainly raising this for discussion about whether it's worthwhile. (fyi @kenzieschmoll @jacob314)

**Comment by kenzieschmoll** [comment]
Is there some utility we could use to generate windows scripts from shell scripts? One downfall of using dart scripts is that we have to setup a Dart SDK on the CI to run these scripts, which can slow things down. In the past @CoderDake has converted some dart tool scripts to shell scripts to avoid this issue.

**Comment by DanTup** [comment]
> One downfall of using dart scripts is that we have to setup a Dart SDK on the CI to run these scripts, which can slow things down.

Good point :-)

Converting them might work. I can see a few scripts people have written that try to do this that work for a very small subset of what bash scripts can do.. Maybe these scripts are simple enough that approach will work. I might give this a try (although I might just write my own replacements in a ps1 script to avoid additional dependencies on things like nodejs).

**Comment by CoderDake** [comment]
@DanTup is there a lot of these scripts that we would need to find solutions for?
If the subset is small, and the things we do with them are simple, then maybe making platform agnostic ones wouldn't be so bad?

**Comment by DanTup** [comment]
Sorry for the delay, I had looked and this and meant to post back...

I had tried to convert these using a script, but didn't get very far. Converting things like the `if` statements automatically was difficult (in part because we have lots of inconsistent ways of writing them across these scripts, but even with those normalised it was tricky).

Perhaps we could use Dart for anything that runs after we know we have a Flutter SDK, and just use `.sh` for the things that run before (such as setting up the SDK)? That might leave less to duplicate/convert into a Windows version at least?

In the first instance, being able to run DevTools from source embedded in VS Code would be useful. I'll make a list of exactly which scripts are needed for this (that are currently `.sh`) and post back.
